outlines/caching.py

```diff
@@ -17,6 +20,28 @@ def cache_fn(fn: Callable):
     return cache_fn
 
 
+def hash_it(*data) -> str:
+    """Pickles and hashes all the data passed to it as args."""
+    result = hashlib.md5()  # nosec B303
+    for datum in data:
+        if isinstance(datum, torch.Tensor):
+            datum = datum.cpu().numpy()
```

Comment:
If we pickle (&hash) without converting to a numpy array, then the final hash may not match as the pickled object also contains reference to tensor's location in memory. Ref: https://github.com/joblib/joblib/issues/1282

Since the tensors might be on cuda device, we copy them to cpu and then convert to numpy.